mmv1/third_party/terraform/resources/resource_bigquery_table.go

```diff
@@ -167,28 +168,32 @@ func bigQueryTableSchemaDiffSuppress(name, old, new string, _ *schema.ResourceDa
 	return eq
 }
 
-func bigQueryTableTypeEq(old, new interface{}) bool {
+func bigQueryTableTypeEq(old, new string) bool {
+	// Do case-insensitive comparison. https://github.com/hashicorp/terraform-provider-google/issues/9472
+	oldUpper := strings.ToUpper(old)
+	newUpper := strings.ToUpper(new)
+
 	equivalentSet1 := []interface{}{"INTEGER", "INT64"}
 	equivalentSet2 := []interface{}{"FLOAT", "FLOAT64"}
 	equivalentSet3 := []interface{}{"BOOLEAN", "BOOL"}
-	eq0 := old == new
-	eq1 := valueIsInArray(old, equivalentSet1) && valueIsInArray(new, equivalentSet1)
-	eq2 := valueIsInArray(old, equivalentSet2) && valueIsInArray(new, equivalentSet2)
-	eq3 := valueIsInArray(old, equivalentSet3) && valueIsInArray(new, equivalentSet3)
+	eq0 := oldUpper == newUpper
+	eq1 := valueIsInArray(oldUpper, equivalentSet1) && valueIsInArray(newUpper, equivalentSet1)
+	eq2 := valueIsInArray(oldUpper, equivalentSet2) && valueIsInArray(newUpper, equivalentSet2)
+	eq3 := valueIsInArray(oldUpper, equivalentSet3) && valueIsInArray(newUpper, equivalentSet3)
 	eq := eq0 || eq1 || eq2 || eq3
 	return eq
 }
 
-func bigQueryTableModeEq(old, new interface{}) bool {
-	equivalentSet := []interface{}{nil, "NULLABLE"}
-	eq0 := old == new
-	eq1 := valueIsInArray(old, equivalentSet) && valueIsInArray(new, equivalentSet)
-	eq := eq0 || eq1
-	return eq
+func bigQueryTableNormalizeMode(mode interface{}) string {
+	if mode == nil {
+		return "NULLABLE"
+	}
+	// Upper-case to get case-insensitive comparisons. https://github.com/hashicorp/terraform-provider-google/issues/9472
+	return strings.ToUpper(mode.(string))
 }
 
-func bigQueryTableModeIsForceNew(old, new interface{}) bool {
-	eq := bigQueryTableModeEq(old, new)
+func bigQueryTableModeIsForceNew(old, new string) bool {
+	eq := old == new
```

Comment:
This is a safe change because we presanitize the value now, right?

Reply:
yeah, since we know that they're both strings (and we've already done string.ToUpper) this is safe.